File: ProjetoSpaceNowAngular/src/app/Models/sala.model.ts
import { AndarModel } from "./andar.model";
import { SalaImagemModel } from "./sala-imagem.model";
import { SalaRecursoModel } from "./sala-recurso.model";
import { SalaResponsavelModel } from "./sala-responsavel.model";
import { SalaTipoModel } from "./sala-tipo.model";

export class SalaModel {
  salaId: number = 0;
  apelido: string = '';
  capacidade: number = 0;
  publicoExterno: boolean = false;
  bitBloqueiaPublicoExterno: boolean = false;
  salaTipoId: number = 0;
  salaTipo: SalaTipoModel = new SalaTipoModel();
  andar: AndarModel = new AndarModel();
  recursos: SalaRecursoModel[] = [];
  responsaveis: SalaResponsavelModel[] = [];
  imagens: SalaImagemModel[] = [];
  ativo: boolean = true;
  aliasCadastro?: string;
  dtCadastro?: string;
  nomeAlteracao?: string;
  aliasAlteracao?: string;
  dtAlteracao?: string;
  ids: string[] = [];
}

File: ProjetoSpaceNow/ProjetoSpaceNow.Api/Models/SalaModel.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace ProjetoSpaceNow.Api.Models
{
    [Table("sala")]
    public class SalaModel : BaseEntity
    {
        [Column("apelido")]
        public string Apelido { get; set; }

        [Column("capacidade")]
        public int Capacidade { get; set; }

        [Column("publico_externo")]
        public bool PublicoExterno { get; set; }

        [Column("andar_id")]
        public int AndarId { get; set; }

        [Reference(typeof(AndarModel))]
        public AndarModel Andar { get; set; }

        [Column("sala_tipo_id")]
        public int SalaTipoId { get; set; }

        [Reference(typeof(SalaTipoModel))]
        public SalaTipoModel SalaTipo { get; set; }

        [Reference(typeof(SalaRecursoModel))]
        public ICollection<SalaRecursoModel> Recursos { get; set; }

        [Reference(typeof(SalaResponsavelModel))]
        public ICollection<SalaResponsavelModel> Responsaveis { get; set; }

        [Reference(typeof(SalaDisponibilidadeModel))]
        public ICollection<SalaDisponibilidadeModel> Disponibilidades { get; set; }
    }
}
